src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5149,7 +5149,27 @@ public override BoundNode VisitUnaryOperator(BoundUnaryOperator node)
         {
             Debug.Assert(!IsConditionalState);
 
-            _ = base.VisitUnaryOperator(node);
+            switch (node.OperatorKind)
+            {
+                case UnaryOperatorKind.BoolLogicalNegation:
+                    VisitCondition(node.Operand);
+                    SetConditionalState(StateWhenFalse, StateWhenTrue);
+                    break;
+                case UnaryOperatorKind.DynamicTrue:
+                    // We cannot use VisitCondition, because the operand is not of type bool.
+                    // Yet we want to keep the result split if it was split.  So we simply visit.
+                    Visit(node.Operand);
+                    break;
+                case UnaryOperatorKind.DynamicFalse:
```

Comment:
>UnaryOperatorKind.DynamicFalse [](start = 21, length = 30)

It feels like this case isn't covered by added tests.

Reply:
Agreed; it would not be particularly useful to do so, since it is only used for the implementation of a user-defined meaning for `||`, for which no reasonable inference can be produced by the caller.  Removing this case.

---
In reply to: [276389190](https://github.com/dotnet/roslyn/pull/34996#discussion_r276389190) [](ancestors = 276389190)